### PR TITLE
Bugfix: Running .optimize() for a RQAOA object not working as expected

### PR DESCRIPTION
**Pull request by raulconchello**
Before, if one optimized twice a RQAOA object, the second time wasn't working as expected. Because the self._q.was compiled with the last problem of the first optimization.  

## TODO
- [x] Every time that .optimize() is run the whole RQAOA process should restart
- [x] Check that now everything works as expected
- [x] Write some test that can check that this is the case

### CODE REVIEW

File: tests/test_workflows.py [hunk 1]
```diff
@@ -655,6 +655,17 @@ def _run_rqaoa(self, type, problem, n_cutoff=5, eliminations=1, p=1, param_type=
 
         return r.results.get_solution()
 
+    def test_rqaoa_complie_multiple_times(self):
```
Comment (vishal-ph) [inline tests/test_workflows.py:658]: *compile typo

File: openqaoa/workflows/optimizer.py [hunk 2]
```diff
@@ -682,7 +682,10 @@ def optimize(self, verbose=False):
         problem = self.problem  
         n_cutoff = self.rqaoa_parameters.n_cutoff
         n_qubits = problem.n
-        counter = self.rqaoa_parameters.counter 
+        counter = self.rqaoa_parameters.counter
+
+        # copy the original qaoa object
+        q = copy.deepcopy(self._q)
```
Comment (vishal-ph) [inline openqaoa/workflows/optimizer.py:688]: perhaps it makes sense to use a more descriptive name for this? `q` does not tell me that this is the QAOA with the original problem.

File: tests/test_workflows.py [hunk 3]
```diff
@@ -655,6 +655,17 @@ def _run_rqaoa(self, type, problem, n_cutoff=5, eliminations=1, p=1, param_type=
 
         return r.results.get_solution()
 
+    def test_rqaoa_compile_multiple_times(self):
```
Comment (vishal-ph) [inline tests/test_workflows.py:658]: should it be `test_rqaoa_optimize_multiple_times`?

File: openqaoa/workflows/optimizer.py (hunk 2 above)
Comment (raulconchello) [reply openqaoa/workflows/optimizer.py:688]: but this q is going to change at each RQAOA step

File: tests/test_workflows.py (hunk 3 above)
Comment (raulconchello) [reply tests/test_workflows.py]: true

File: openqaoa/workflows/optimizer.py (hunk 2 above)
Comment (raulconchello) [reply openqaoa/workflows/optimizer.py:688]: Exactly

**Review by vishal-ph — APPROVED** [review]

**Pull request merged**